relayer/processor/path_end_runtime.go

```diff
@@ -933,36 +918,92 @@ func (pathEnd *pathEndRuntime) trackProcessingMessage(tracker messageToTrack) ui
 		}
 		msgProcessCache, ok := pathEnd.connProcessing[eventType]
 		if !ok {
-			msgProcessCache = make(connectionKeySendCache)
+			msgProcessCache = newConnectionKeySendCache()
 			pathEnd.connProcessing[eventType] = msgProcessCache
 		}
 
-		if inProgress, ok := msgProcessCache[connectionKey]; ok {
-			retryCount = inProgress.retryCount + 1
-		}
-
-		msgProcessCache[connectionKey] = processingMessage{
-			lastProcessedHeight: pathEnd.latestBlock.Height,
-			retryCount:          retryCount,
-			assembled:           t.assembled != nil,
+		inProgress := msgProcessCache.get(connectionKey)
+		if inProgress == nil {
+			msgProcessCache.set(connectionKey, pathEnd.latestBlock.Height, t.assembled != nil)
+		} else {
+			inProgress.setProcessing(pathEnd.latestBlock.Height, t.assembled != nil, inProgress.retryCount+1)
 		}
 	case clientICQMessageToTrack:
 		queryID := t.msg.info.QueryID
 
-		if inProgress, ok := pathEnd.clientICQProcessing[queryID]; ok {
-			retryCount = inProgress.retryCount + 1
-		}
-
-		pathEnd.clientICQProcessing[queryID] = processingMessage{
-			lastProcessedHeight: pathEnd.latestBlock.Height,
-			retryCount:          retryCount,
-			assembled:           t.assembled != nil,
+		inProgress := pathEnd.clientICQProcessing.get(queryID)
+		if inProgress == nil {
+			pathEnd.clientICQProcessing.set(queryID, pathEnd.latestBlock.Height, t.assembled != nil)
+		} else {
+			inProgress.setProcessing(pathEnd.latestBlock.Height, t.assembled != nil, inProgress.retryCount+1)
 		}
 	}
 
 	return retryCount
 }
 
+func (pathEnd *pathEndRuntime) trackFinishedProcessingMessage(tracker messageToTrack) {
+	switch t := tracker.(type) {
+	case packetMessageToTrack:
+		eventType := t.msg.eventType
+		sequence := t.msg.info.Sequence
+		channelKey, err := t.msg.channelKey()
+		if err != nil {
```

Comment:
we are silently ignoring an error here, does it make sense to return an error from this method? it looked like this is called after broadcasting the message could an error here result in the problematic message never being purged from the cache?

Reply:
we can silently ignore the error here because if we observe a completion event (e.g. ack, timeout, chanopenconfirm, etc.), then the caches will be cleared and we no longer need to track the processing status.